presidential-elections/utils/model.py

```diff
@@ -581,239 +847,15 @@ def _build_data_containers(
-                election_party_time_coefs,
-                axes=(1, 0),
-            ),
-            dims=("countdown", "parties_complete", "elections"),
-        )
-
-    @staticmethod
-    def _build_regressions(
-        party_intercept: pm.Distribution,
-        election_party_intercept: pm.Distribution,
-        party_time_effect: pm.Distribution,
-        election_party_time_effect: pm.Distribution,
-        unemployment_effect: pm.Distribution,
-        poll_bias: pm.Distribution,
-        house_effects: pm.Distribution,
-        house_election_effects: pm.Distribution,
-        data_containers: Dict[str, pm.Data],
-        non_competing_parties: Dict[str, np.ndarray],
-    ) -> Tuple[pm.Distribution, pm.Distribution]:
-
-        # regression for polls
-        latent_mu = (
-            party_intercept
-            + election_party_intercept[data_containers["election_idx"]]
-            + party_time_effect[data_containers["countdown_idx"]]
-            + election_party_time_effect[
-                data_containers["countdown_idx"], :, data_containers["election_idx"]
-            ]
-            + aet.dot(
-                data_containers["stdz_unemp"][:, None], unemployment_effect[None, :]
-            )
-        )
-        latent_mu = latent_mu + non_competing_parties["polls_additive"]
-        pm.Deterministic(
-            "latent_popularity",
-            aet.nnet.softmax(latent_mu),
-            dims=("observations", "parties_complete"),
-        )
-        noisy_mu = (
-            latent_mu
-            + poll_bias[None, :]  # let bias vary during election period?
-            + house_effects[data_containers["pollster_idx"]]
-            + house_election_effects[
-                data_containers["pollster_idx"], :, data_containers["election_idx"]
-            ]
-            * non_competing_parties["polls_multiplicative"]
-        )
-
-        # regression for results
-        latent_mu_t0 = (
-            party_intercept
-            + election_party_intercept
-            + party_time_effect[0]
-            + election_party_time_effect[0].T
-            + aet.dot(
-                data_containers["election_unemp"][:, None], unemployment_effect[None, :]
-            )
-        )
-        latent_mu_t0 = latent_mu_t0 + non_competing_parties["results"]
-
-        return (
-            pm.Deterministic(
-                "noisy_popularity",
-                aet.nnet.softmax(noisy_mu),
-                dims=("observations", "parties_complete"),
-            ),
-            pm.Deterministic(
-                "latent_pop_t0",
-                aet.nnet.softmax(latent_mu_t0),
-                dims=("elections", "parties_complete"),
-            ),
-        )
-
     def sample_all(
         self, *, model: pm.Model = None, var_names: List[str], **sampler_kwargs
     ) -> arviz.InferenceData:
         """
         Sample the model and return the trace.
+
+        TODO: Add 3 distinct functions to sample from the prior predictive,
```

Comment:
I think you can just add kwargs here

Reply:
Rémi not like kwargs when he can avoid them. We can keep this function, however.